torch/_inductor/ir.py

```diff
@@ -306,6 +306,21 @@ def realize(self):
         """
         raise NotImplementedError(f"realize NYI on {type(self)}")
 
+    # The abstract method declarations below serve to convince mypy that all IRNode instances have these functions
+    # defined, while having no effect at runtime. We cannot create stub implementations here because other parts of
+    # the code dynamically check for defined attributes.
+    get_device: Callable[[], torch.device]
+    get_dtype: Callable[[], torch.dtype]
+    get_name: Callable[[], str]
+    get_reads: Callable[[], Any]
+    get_stride: Callable[[], Any]
+    get_storage_numel: Callable[[], Any]
+    has_exceeded_max_reads: Callable[[], bool]
+    make_loader: Callable[[], Callable[[Any], Any]]
+    make_indexer: Callable[[], Callable[[Any], Any]]
+    mark_reuse: Callable[[List[Any]], None]
+    realize_hint: Callable[[], None]
```

Comment:
Ideally I think we should have stricter typing and not use `IRNode` everywhere, but this is still much better than before so I'm happy to have a workaround for now.